### ISSUE TITLE
Operator (4.5.0) "Observed a panic: "invalid memory address or nil pointer dereference" when creating tenant

### ISSUE THREAD

**Issue by enrico2828** [minio/operator#1294]
We upgraded OpenShift Minio Operator to version 4.5.0 and since then we are unable to create any tentant at all. The operator pod crashes with Runtime error.

## Expected Behavior
Tenant should be created.

## Current Behavior
Operator crashed.
```
    ~  k logs minio-operator-585b94b97f-f7vnk                ✔  6s   ocp201/openshift-operators ⎈  10:17:06 
I0919 08:12:10.498743       1 main.go:70] Starting MinIO Operator
I0919 08:12:10.978042       1 main.go:169] caBundle on CRD updated
I0919 08:12:10.978523       1 main-controller.go:244] Setting up event handlers
I0919 08:12:10.978685       1 leaderelection.go:243] attempting to acquire leader lease openshift-operators/minio-operator-lock...
I0919 08:12:10.984072       1 main-controller.go:497] new leader elected: minio-operator-585b94b97f-kkz9v
I0919 08:13:14.951333       1 leaderelection.go:253] successfully acquired lease openshift-operators/minio-operator-lock
I0919 08:13:14.951531       1 main-controller.go:478] minio-operator-585b94b97f-f7vnk: I've become the leader
I0919 08:13:14.951629       1 main-controller.go:390] Waiting for API to start
I0919 08:13:14.951687       1 main-controller.go:382] Starting HTTP Upgrade Tenant Image server
I0919 08:13:16.148274       1 request.go:655] Throttling request took 1.189364609s, request: GET:https://172.30.0.1:443/apis/direct.csi.min.io/v1beta3?timeout=32s
I0919 08:13:21.947890       1 main-controller.go:351] Using Kubernetes CSR Version: v1
I0919 08:13:21.958088       1 main-controller.go:355] Starting HTTPS API server
I0919 08:13:21.958182       1 main-controller.go:393] Waiting for Upgrade Server to start
I0919 08:13:21.958195       1 main-controller.go:397] Starting Tenant controller
I0919 08:13:21.958200       1 main-controller.go:400] Waiting for informer caches to sync
I0919 08:13:21.958208       1 main-controller.go:405] Starting workers
I0919 08:13:21.966455       1 monitoring.go:100] 'na-minio/storage-lite' no pool is initialized
I0919 08:13:21.970228       1 upgrades.go:91] Upgrading v4.5
E0919 08:13:21.989477       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 935 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1677a60?, 0x26253f0})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:74 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0002faad0?})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1677a60, 0x26253f0})
	runtime/panic.go:838 +0x207
github.com/minio/operator/pkg/controller/cluster.(*Controller).checkForUpgrades(0xc0000fca20, {0x1aeade8, 0xc000128000}, 0xc0015c4800)
	github.com/minio/operator/pkg/controller/cluster/upgrades.go:93 +0xae2
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc0000fca20, {0xc000f56060, 0x15})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:629 +0x2f1
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x16064c0?, 0xc0002faad0})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:571 +0x2e6
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc0000fca20)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:583 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0xc003db46a0?)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:523 +0x47
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x1ad4720, 0xc00288bc80}, 0x1, 0xc0001143c0)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:408 +0x3f8
E0919 08:13:21.989548       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 935 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1677a60?, 0x26253f0})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:74 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1677a60, 0x26253f0})
	runtime/panic.go:838 +0x207
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0002faad0?})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1677a60, 0x26253f0})
	runtime/panic.go:838 +0x207
github.com/minio/operator/pkg/controller/cluster.(*Controller).checkForUpgrades(0xc0000fca20, {0x1aeade8, 0xc000128000}, 0xc0015c4800)
	github.com/minio/operator/pkg/controller/cluster/upgrades.go:93 +0xae2
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc0000fca20, {0xc000f56060, 0x15})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:629 +0x2f1
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x16064c0?, 0xc0002faad0})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:571 +0x2e6
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc0000fca20)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:583 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0xc003db46a0?)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:523 +0x47
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x1ad4720, 0xc00288bc80}, 0x1, 0xc0001143c0)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:408 +0x3f8
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1474ea2]

goroutine 935 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1677a60, 0x26253f0})
	runtime/panic.go:838 +0x207
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0002faad0?})
	k8s.io/apimachinery@v0.20.6/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1677a60, 0x26253f0})
	runtime/panic.go:838 +0x207
github.com/minio/operator/pkg/controller/cluster.(*Controller).checkForUpgrades(0xc0000fca20, {0x1aeade8, 0xc000128000}, 0xc0015c4800)
	github.com/minio/operator/pkg/controller/cluster/upgrades.go:93 +0xae2
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc0000fca20, {0xc000f56060, 0x15})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:629 +0x2f1
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x16064c0?, 0xc0002faad0})
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:571 +0x2e6
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc0000fca20)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:583 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0xc003db46a0?)
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:523 +0x47
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x1ad4720, 0xc00288bc80}, 0x1, 0xc0001143c0)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	k8s.io/apimachinery@v0.20.6/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
	github.com/minio/operator/pkg/controller/cluster/main-controller.go:408 +0x3f8
```
## Possible Solution
No solution was found, we tried all kinds of different variations in tenant, but nothing.

## Steps to Reproduce (for bugs)

1. Install the Operator 4.5.0 on OpenShift
2. Create a tenant, like so:
```
apiVersion: minio.min.io/v2
kind: Tenant
metadata:
  annotations:
    prometheus.io/path: /minio/v2/metrics/cluster
    prometheus.io/port: "9000"
    prometheus.io/scrape: "true"
  labels:
    app: minio
  name: test-minio
  namespace: na-minio
spec:
  # minio root account credentials, please ensure it exists
  credsSecret:
    name: minio-root-user-creds
  env:
  # set erasure coding
  - name: MINIO_STORAGE_CLASS_STANDARD
    value: "EC:2"
  externalCaCertSecret: []
  externalCertSecret:
    - name: na-minio-certmanager-tls
      type: kubernetes.io/tls
  image: >-
    quay.io/minio/minio@sha256:3f9b739b3903d6022f8c5309991b199d7006b802a31eef42eeec615005eb765f
  imagePullSecret: {}
  mountPath: /export
  log:
    affinity:
      nodeAffinity: {}
      podAffinity: {}
      podAntiAffinity: {}
    annotations: {}
    audit:
      diskCapacityGB: 7000
    db:
      affinity:
        nodeAffinity: {}
        podAffinity: {}
        podAntiAffinity: {}
      annotations: {}
      image: ""
      initimage: ""
      labels: {}
      nodeSelector: {}
      resources: {}
      securityContext:
        fsGroup: 1000870000
        runAsGroup: 1000870000
        runAsNonRoot: false
        runAsUser: 1000870000
      serviceAccountName: minio-db
      tolerations:
      - key: "node-role.kubernetes.io/minio"
        operator: "Exists"
        effect: "NoSchedule"
      volumeClaimTemplate:
        metadata: {}
        spec:
          accessModes:
          - ReadWriteOnce
          resources:
            requests:
              storage: 7000Gi
          storageClassName: minio-local-storage
  podManagementPolicy: Parallel
  pools:
  - name: pool-0
    servers: 4
    volumeClaimTemplate:
      metadata:
        name: data
      spec:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: 7451Gi
        storageClassName: minio-local-storage
    securityContext:
      seLinuxOptions:
        type: "spc_t"
      fsGroup: 1000870000
      runAsGroup: 1000870000
      runAsNonRoot: true
      runAsUser: 1000870000
    volumesPerServer: 2
    tolerations:
    - key: "node-role.kubernetes.io/minio"
      operator: "Exists"
      effect: "NoSchedule"
  priorityClassName: ""
  prometheusOperator: false
  requestAutoCert: false
  s3:
    bucketDNS: false
  serviceAccountName: "minio-selinux"
  serviceMetadata:
    consoleServiceAnnotations: {}
    consoleServiceLabels: {}
    minioServiceAnnotations: {}
    minioServiceLabels: {}
  subPath: ""
  users:
  - name: storage-user
```

4. Observe the operator crash.

## Context
It is now not possible to install Minio on the OpenShift cluster. 

## Regression
With previous version of the operator, tenant installed fine.

## Your Environment

* Version used (`minio-operator`): 4.5.0
* Environment name and version (e.g. kubernetes v1.17.2): OpenShift 4.9.41
* Server type and version: Baremetal servers HPE
* Operating System and version (`uname -a`): CoreOS
* Link to your deployment file:

**Comment by DanielMemmelAa** [comment]
Same issue here during Tenant creation when Upgrading v4.5 is called.

- Minio operator: v4.5
- Environment name and version: Kubernetes 1.21
- OS: Ubuntu 20.04.1

**Comment by dvaldivia** [comment]
Will get fixed by #1296

**Comment by AljoschaP** [comment]
Its working again :smile_cat: Thank you :bow:
